internal/ext/iterx/iterx.go

```diff
@@ -170,3 +170,19 @@ func Join[T any](seq iter.Seq[T], sep string) string {
 	})
 	return out.String()
 }
+
+// Chain returns an iterator that calls a sequence of iterators in sequence.
```

Comment:
I usually see this function called `Flatten`.

Reply:
I would have called it Flatten if it took an iterator of iterators.